types/base/api.d.ts

```diff
@@ -0,0 +1,57 @@
+type Part = {
+    name: string;
+    material: string;
+    color: string;
+}
+
+type InitialsExtra = Record<string, {
+    initials: string;
+    engraving?: string;
+}>
+
+type Spec = {
+    brand: string;
+    model: string;
+    parts: Part[];
+    initials: string;
+    engraving: string;
+    initialsExtra: InitialsExtra;
+    variant?: string;
+    version?: string;
+    description?: string;
+};
+
+type ImportOrderOptions = {
+    brand?: string;
+    model?: string;
+    variant?: string;
+    parts?: Part[];
+    initials?: string;
+    engraving?: string;
+    initialsExtra?: InitialsExtra;
+    gender?: string;
+    size?: number;
+    pending?: boolean;
+    notify?: boolean;
+    productId?: number;
+    currency?: string;
+    country?: string;
+    meta?: string[];
+}
+
+type RequestOptions = {
+    url?: string;
+    method?: string;
+    params?: Record<string, unknown>,
+    headers?: Record<string, unknown>,
+    auth?: boolean
+}
+
+export declare class RipeAPI {
+    key?: string;
+
+    constructor(options?: any);
```

Comment:
For some reason we're using `unknown` everywhere so stick to it.

Reply:
My bad. `unknow` is better because when using it forces the user to cast it to the desired type.